packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -29,4 +29,37 @@ export class GOVUKFrontendComponent {
       throw new SupportError()
     }
   }
+
+  /**
+   *
+   * @protected
+   * @param {Schema} schema
+   * @param {{ [key: string]: number | string | object | undefined}} config
+   * @returns {boolean} Whether the config is valid
+   */
+  validateConfig(schema, config) {
+    if (schema.anyOf) {
+      // Check if at least one of the conditions in anyOf is met
+      for (const condition of schema.anyOf) {
+        const requiredProps = condition.required
+        if (requiredProps.every((prop) => prop in config && config[prop])) {
+          return true
+        }
+      }
+      return false
+    }
+    return true
+  }
 }
+
+/**
+ * @typedef {object} Schema
+ * @property {SchemaItem[]} [anyOf] - An array of conditions
+ * @property {*} [otherProperties] - Other properties that may exist in the schema.
```

Comment:
Mind if we drop `otherProperties` if it's unused? Save it for the PR that adds more properties